tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js

```diff
@@ -156,5 +156,101 @@ describe('aws-sdk test put bucket policy', () => {
+            const params = getPolicyParams({
+                key: 'Condition', value: {
+                    NumericGreaterThanEquals: {
+                        's3:object-lock-remaining-retention-days': '30',
+                    },
+                },
+            });
+            s3.putBucketPolicy(params, err => assertError(err, null, done));
+        });
+
+        // yep, this is the expected behaviour
+        it('should not reject policy with invalid s3:object-lock-remaining-retention-days value', done => {
+            const params = getPolicyParams({
+                key: 'Condition', value: {
+                    NumericGreaterThanEquals: {
+                        's3:object-lock-remaining-retention-days': '-1', // Invalid value
+                    },
+                },
+            });
+            s3.putBucketPolicy(params, err => assertError(err, null, done));
+        });
+
+        // this too ¯\_(ツ)_/¯
+        it('should not reject policy with a key starting with aws:', done => {
+            const params = getPolicyParams({
+                key: 'Condition', value: {
+                    NumericGreaterThanEquals: {
+                        'aws:have-a-nice-day': 'blabla', // Invalid value
+                    },
+                },
+            });
+            s3.putBucketPolicy(params, err => assertError(err, null, done));
+        });
+
+        it('should reject policy with a key that does not exist that does not start with aws:', done => {
+            const params = getPolicyParams({
+                key: 'Condition', value: {
+                    NumericGreaterThanEquals: {
+                        'have-a-nice-day': 'blabla', // Invalid value
+                    },
+                },
+            });
+            s3.putBucketPolicy(params, err => assertError(err, 'MalformedPolicy', done));
+        });
+
+        it('should enforce policies with both SourceIp and s3:object-lock conditions together', done => {
+            const params = getPolicyParams({
+                key: 'Condition', value: {
+                    IpAddress: {
+                        'aws:SourceIp': '192.168.100.0/24',
+                    },
+                    NumericGreaterThanEquals: {
+                        's3:object-lock-remaining-retention-days': '30',
+                    },
+                },
+            });
+            s3.putBucketPolicy(params, err => assertError(err, null, done));
+        });
+
+        it('should return error if a condition one of the condition values is invalid', done => {
+            const params = getPolicyParams({
+                key: 'Condition', value: {
+                    IpAddress: {
+                        'aws:SourceIp': '192.168.100',
+                    },
+                    NumericGreaterThanEquals: {
+                        's3:object-lock-remaining-retention-days': '30',
```

Comment:
Question:
Where is the logic to test this?
From what I understand this condition allowed client with IP '192.168.100' to access resources which has remaining-retention-days as 30.

When a user tries to access such an object, where do we check this and deny the request?

Reply:
We should add a test for the logic in integration, which puts object lock retention for a period and tests whether a user is able to access or not access the resource.
I will take a note on it.
But I am still not sure where the logic to check that is. Can you help guide me?

Reply:
> Where is the logic to test the 30 day?

Here, we're not testing the logic of the condition values and whether it makes sense to set such conditions, here we're simply testing the ability to set them; functionality of such conditions is for integration testing. And yes, supposing we have a valid IP here, in a real scenario, this would limit the call on an object to the given IP to objects that have 30 or more days retention left.

> But I am still not sure where the logic to check that is. Can you help guide me?
Do you mean where in the code we test this behaciour? Or do you mean the coherency of such a test case? In Integration, we would have test cases that have more realistic use cases I'm sure.

Reply:
> When a user tries to access such an object, where do we check this and deny the request?

Integration, as this is a bucket policy check question

Reply:
I meant, 

If bucket policy is set with a condition for retention days of  less than 30
where in the code do we check if the retention day is less than 30 and if not deny the request.
For example, IAM policies had the logic already to verify the IP address.

I am not talking about a test with mocha, I am talking about authorization logic of the API request. Agreed the test will go in integration.

Reply:
Reference: Description of https://scality.atlassian.net/browse/S3C-3580
> Update bucket policy to support s3:object-lock-remaining-retention-days condition
The above is done


> Update putBucketObjectLockConfig to check for the bucket policy and prevent setting retention period outside of the allowed limit of bucket policy
This we need to do in a PR